python/src/scipp/typing.py

```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Neil Vaytet
+
+import typing as _std_typing
+
+from ._scipp import core as sc
+
+
+def is_variable(obj: _std_typing.Any) -> bool:
+    """
+    Return True if the input is of type scipp.Variable.
+    """
+    return isinstance(obj, sc.Variable)
+
+
+def is_scalar(obj: _std_typing.Any) -> bool:
+    """
+    Return True if the input is a scalar.
+    """
+    return obj.dims == []
+
+
+def is_dataset(obj: _std_typing.Any) -> bool:
+    """
+    Return True if the input is of type scipp.Dataset.
+    """
+    return isinstance(obj, sc.Dataset)
+
+
+def is_data_array(obj: _std_typing.Any) -> bool:
```

Comment:
I wonder if all these functions are still useful? When we still had all the `*View` types they were since they avoid checking for multiple things, but now I feel using `isinstance` directly may be clearer?

Reply:
True.
Those functions would in principle allow us to change the implementation again and support multiple different Variable classes without changing these kinds of check in user code. But I don't think that is a strong enough argument. Would you agree?

Reply:
Agree. Unlikely that we change back.